### ISSUE TITLE
Rule Request: no_duplicated_condition

### ISSUE THREAD

**Issue by lo1tuma** [realm/SwiftLint#4666]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

1. Why should this rule be added?

This rule should flag conditions that are used multiple times in the same branching statement like if-else-if or switch statement.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

Violating Examples

```swift
if x < 5 {
  foo()
} else if x < 2 {
  bar()
} else if x < 5 { // duplicated condition
  baz()
}

switch x {
case "a":
  foo()
case "a" // duplicated condition
  bar()
}
```

Non-Violating Examples

```swift
if x < 5 {
  foo()
  baz()
} else if x < 2 {
  bar()
}

switch x {
case "a":
  foo()
  bar()
}
```

3. Should the rule be configurable, if so what parameters should be configurable?

This rule doesn’t need to be configurable.

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

This would be a good candidate for a builtin rule as it points out bad structured code.
